move trigger tab to react

sunilyadav840 opened the pull request:
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/855?feature.someFeatureFlagYouMightNeed=true)

Migration of trigger Tab into react and remove knockout TriggerTab.ts and TriggerTab.html file

https://user-images.githubusercontent.com/79906609/120454077-2f973b00-c3b1-11eb-84b8-8afd633733ea.mp4


https://user-images.githubusercontent.com/79906609/121141252-66110200-c858-11eb-991b-dd193d3d1ef5.mp4

southpolesteve commented on src/Explorer/Tabs/TriggerTabContent.tsx:
```diff
@@ -0,0 +1,339 @@
+      triggerOperation: triggerOperation as SqlTriggerResource["triggerOperation"],
+      triggerType: triggerType as SqlTriggerResource["triggerType"],
+    })
+      .then(
+        (createdResource) => {
```
Switch this to use async/await

sunilyadav840 replied:
Updated code

southpolesteve commented:
Bug: After saving a Tigger, the ID should no longer be editable

southpolesteve commented:
Bug: Viewing an existing trigger, the ID should not be editable

sunilyadav840 commented on src/Explorer/Tree/Trigger.ts:
```diff
@@ -40,7 +40,7 @@ export default class Trigger {
   }
 
-  public static create(source: ViewModels.Collection, event: MouseEvent) {
+  public static create(container: Explorer, source: ViewModels.Collection, event: MouseEvent) {
```
Removed.

sunilyadav840 commented:
@southpolesteve 

> Bug: After saving a Tigger, the ID should no longer be editable

Fixed the bug. Added video link in description for reference.

sunilyadav840 commented:
> Bug: Viewing an existing trigger, the ID should not be editable

Fixed the bug. Added video link in description for reference

victor-meng commented:
The browser tab starts to flicker after opening the trigger tab.

https://user-images.githubusercontent.com/56978073/121273793-e7c75680-c87d-11eb-8a5a-e2fcd1ccecd0.mp4

victor-meng commented on src/Explorer/Tabs/TriggerTabContent.tsx:
```diff
@@ -0,0 +1,343 @@
+            placeholder="Enter the new trigger id"
+            size={40}
+            value={triggerId}
+            disabled={!isIdEditable}
```
Use `readOnly` instead of `disabled`.

sunilyadav840 replied:
Changes updated

sunilyadav840 commented:
> The browser tab starts to flicker after opening the trigger tab. Check out 0:32 and 0:38 in the video below.
> 
>  vimeng-sql-test.-.Microsoft.Azure.-.Google.Chrome.2021-06-08.17-19-53.mp4 
> After this icon shows up in the editor, the tab becomes very laggy. If I leave the tab open for some time then the browser tab would eventually crash.
> ![image](https://user-images.githubusercontent.com/56978073/121274477-6ec8fe80-c87f-11eb-9ace-bb0d97c8b526.png)

Thank you @victor-meng  for mentioning issue.
It is fixed and code pushed. 
Uploaded video for reference.

https://user-images.githubusercontent.com/79906609/121860089-22b20a00-cd16-11eb-8b3d-d9e95fa79e6d.mp4
